### PR TITLE
Fix group

### PR DESCRIPTION
**Pull request by maasencioh**
Group method brok in the case when reached a leaf

What do you think @mljs/collaborators ?

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -1,5 +1,6 @@
 language: node_js
 node_js:
-  - "4"
-  - "6"
-  - "stable"
+  - 4
+  - 5
```
Comment (targos) [inline .travis.yml]: 6 was correct. Node v5 is not supported anymore

**Comment by targos** [comment]
Can you add a test that fails before the fix and works after?

**Comment by maasencioh** [comment]
@targos the test case is in the latest commit

**Comment by targos** [comment]
LGTM if the result is correct

**Pull request merged**